README.md

```diff
@@ -54,7 +61,14 @@ repositories {
 }
 
 dependencies {
-    implementation 'com.github.xmartlabs:fountain:0.3.0'
+    // This dependency is required only if you want to use a Retrofit service without a special adapter. 
+    implementation 'com.github.xmartlabs.fountain:fountain-retrofit:0.3.0'
+
+    // This dependency is required only if you want to use a Coroutine retrofit adapter.
+    implementation 'com.github.xmartlabs.fountain:fountain-coroutines:0.3.0'
+
+    // This dependency is required only if you want to use a RxJava2 retrofit adapter.
+    implementation 'com.github.xmartlabs.fountain:fountain-rx2:0.3.0'
 }
```

Comment:
What if we want to use multiple adapters? Isn't a core code that will be imported twice? Will there be a conflict then?

Reply:
No, no problem with that. For example if you want to include push notifications and deeplinks in your app using firebase, both dependencies will include the firebase core dependency. Here is the same. makes sense?

Reply:
> No, no problem with that. For example if you want to include push notifications and deeplinks in your app using firebase, both dependencies will include the firebase core dependency. Here is the same. makes sense?

Yeah. So either the three import a core library transitively (the core is not just code but a library so there is reuse and not duplication). Is the core the first one of the 3? Or the core is not listed?

Reply:
Yep, exactly. The core is not listed, but you can add it using `implementation 'com.github.xmartlabs.fountain:fountain-core:0.X.Y'`. Furthermore, if you take a look at the other modules dependencies, the core is included as an [`api` dependency](https://github.com/xmartlabs/fountain/blob/b396ab3b6580fe4325c4d37463930c0364bb2b07/fountain-coroutines/build.gradle#L35), that means that you can use the core stuff without declaring that dependency.

Reply:
I don't know, for me it doesn't make much sense. I mean, it's an internal module, for example if you want to include the firebase push notification module, the documentation say that you have to import [`implementation 'com.google.firebase:firebase-messaging:17.3.3'`](https://firebase.google.com/docs/cloud-messaging/android/client), it doesn't talk about the core module. However, I know that it exists!

Reply:
Alright!